ztp/tools/pgt2acm/packages/labels/labels.go

```diff
@@ -0,0 +1,67 @@
+package labels
+
+import (
+	"encoding/json"
+	"fmt"
+
+	yamlconv "github.com/ghodss/yaml"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/selection"
+	yaml "sigs.k8s.io/yaml/goyaml.v3"
+)
+
+// Converts a label list and an exclude label list to a Selector
+func LabelToSelector(labelList, excludeLabelList map[string]string) (selector labels.Selector) {
+	selector = labels.NewSelector()
+	selector = addRequirements(labelList, selector, false)
+	selector = addRequirements(excludeLabelList, selector, true)
+	return selector
+}
+
+// converts a labe list to a list of requirements in a selector
+func addRequirements(labelList map[string]string, selector labels.Selector, exclude bool) labels.Selector {
+	for key, value := range labelList {
+		if key == "" {
+			continue
+		}
+		if exclude {
+			requirements, _ := labels.NewRequirement(key, selection.DoesNotExist, []string{value})
+			selector = selector.Add(*requirements)
+			continue
+		}
+		if value == "" {
+			requirements, _ := labels.NewRequirement(key, selection.Exists, []string{})
+			selector = selector.Add(*requirements)
+			continue
+		}
+		if value != "" {
+			requirements, _ := labels.NewRequirement(key, selection.In, []string{value})
+			selector = selector.Add(*requirements)
+			continue
+		}
+	}
+	return selector
+}
+
+// outputs the Selector as a generic Yaml
```

Comment:
follow the convention to describe func name. This is a comment in general. Please verify for all function name description.

Reply:
Fixed